server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalGeoCentroid.java

```diff
@@ -84,12 +76,27 @@ protected InternalGeoCentroid copyWith(double firstSum, double secondSum, long t
     }
 
     @Override
-    public InternalAggregation finalizeSampling(SamplingContext samplingContext) {
-        return new InternalGeoCentroid(name, centroid, samplingContext.scaleUp(count), getMetadata());
+    protected String nameFirst() {
+        return "lat";
+    }
+
+    @Override
+    protected double extractFirst(SpatialPoint point) {
+        return point.getY();
+    }
+
+    @Override
+    protected String nameSecond() {
+        return "lon";
+    }
+
+    @Override
+    protected double extractSecond(SpatialPoint point) {
+        return point.getX();
     }
 
-    static class Fields {
-        static final ParseField CENTROID_LAT = new ParseField("lat");
-        static final ParseField CENTROID_LON = new ParseField("lon");
+    @Override
+    public InternalAggregation finalizeSampling(SamplingContext samplingContext) {
```

Comment:
This method can be deleted, or alternatively delete the copyWith method. Seems I added duplicates unnecessarily here.